lib/document.js

```diff
@@ -580,17 +580,12 @@ Document.prototype._batchSave = function(executeInsert) {
  * @return {Promise}
  */
 Document.prototype._batchSaveSelf = function(executeInsert) {
-  var model = this._getModel();
+  var self = this;
 
   return new Promise(function(resolve, reject) {
-    if (model._tableReady === true) {
+    self.constructor.ready().then(function() {
```

Comment:
Thinky has a really special prototype chain. And by special, I mean unique.
`self.constructor` is `self.getModel()`, not `self._getModel()`